### PR TITLE
Postgres AD auth helper

### PR DESCRIPTION
**Pull request by marcin-dardzinski**

### CODE REVIEW

File: src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+namespace LeanCode.Npgsql.ActiveDirectory;
+
+public static class NpgsqlDataSourceBuilderExtensions
+{
+    public static NpgsqlDataSourceBuilder UseAzureActiveDirectoryAuthentication(
+        this NpgsqlDataSourceBuilder builder,
+        TokenCredential credential
+    )
+    {
+        return builder.UsePeriodicPasswordProvider(
```
Comment (jakubfijalkowski) [inline src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs:15]: Can we have a test if this works? :P 

Yeah, yeah, I know, but I think it is high time to have also Postgres-specific tests.

File: src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            async (_, ct) =>
+            {
+                var token = await credential.GetTokenAsync(
+                    new(new[] { "https://ossrdbms-aad.database.windows.net" }, null),
```
Comment (Saancreed) [inline src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs]: Locally some analyzer complained to me that this `new[] { "https://ossrdbms-aad.database.windows.net" }` should be a `private static readonly string[]` field for performance reasons.

**Review by Saancreed — APPROVED** [review]

File: src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs (hunk 2 above)
Comment (marcin-dardzinski) [reply src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs]: Yeah I saw it too, but TBH I don't have any guarantees regarding how this array is consumed. I _expect_ it not to be modified, but I'm not sure.

File: src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs (hunk 1 above)
Comment (marcin-dardzinski) [reply src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs:15]: What kind of test do you expect? A unit (say non-azure) one with some mocked `TokenCredential` or a E2E one? 
I can think of a test skipped by default, unless you set up credentials with some provisioning script to be run if necessary.

Comment (jakubfijalkowski) [reply src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs:15]: If you can develop a unit test, then unit test. If you can't - then integration test that is run on CI will be OK.

File: src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs (hunk 2 above)
Comment (jakubfijalkowski) [reply src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs]: I think we can safely assume that :)

File: src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs (hunk 1 above)
Comment (marcin-dardzinski) [reply src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs:15]: I did a e2e test connecting to postgres but it requires a database. We can skip it just like other (send grid/fcm/smsapi) tests and run it only when we touch the integration, or I can set up all the resources in the cloud and run it every time.

Comment (jakubfijalkowski) [reply src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs:15]: We should have that on CI (separate workflow) and probably have two targets in the local flow.

**Comment by marcin-dardzinski** [comment]
Added cloud tests in github actions

File: .github/workflows/core_build_publish.yml [hunk 3]
```diff
@@ -133,6 +133,11 @@ jobs:
       - name: Test
         run: dotnet msbuild /t:RunTestsWithCoverage
         working-directory: test
+        env:
```
Comment (jakubfijalkowski) [inline .github/workflows/core_build_publish.yml:139]: Can we run the cloud/azure tests as a separate job? Will be easier to diagnose.

**Review by jakubfijalkowski — APPROVED** [review]

**Pull request merged**